crates/bevy_ui/src/lib.rs

```diff
@@ -184,19 +186,33 @@ impl Plugin for UiPlugin {
                 ui_layout_system
                     .in_set(UiSystem::Layout)
                     .before(TransformSystem::TransformPropagate),
-                resolve_outlines_system
-                    .in_set(UiSystem::Outlines)
-                    .after(UiSystem::Layout)
-                    // clipping doesn't care about outlines
-                    .ambiguous_with(update_clipping_system)
-                    .ambiguous_with(widget::text_system),
-                ui_stack_system
-                    .in_set(UiSystem::Stack)
-                    // the systems don't care about stack index
-                    .ambiguous_with(update_clipping_system)
-                    .ambiguous_with(resolve_outlines_system)
-                    .ambiguous_with(ui_layout_system)
-                    .ambiguous_with(widget::text_system),
+                {
+                    #[allow(clippy::let_and_return)]
```

Comment:
We should be able to avoid the let_and_return pattern by using app.configure_sets to do the feature-gated config.

Reply:
I've never used `configure_sets` before, so I'm unsure of how this would be accomplished. Are you suggesting something like this:

```rust
app.add_systems(
    PostUpdate,
    ( ... ),
);

#[cfg(feature = "bevy_text")]
app.configure_sets(
    PostUpdate,
    resolve_outlines_system.ambiguous_with(widget::text_system),
);
```

Reply:
Yep, that's exactly the idea :)

Reply:
I've been trying but have been unable to get this to compile. What am I doing wrong?

https://gist.github.com/BD103/0519f7b9f7274764d363880ab65db401

Reply:
Ah. You may need to give that system an explicit system set label and then use that.

Reply:
#10858 would eliminate this distinction.

Reply:
Ok, I added a system set in 4f40ccfe0d0b2fce126812bf5aae7cd0513ae2fb. It's less than pretty, but what do you think?